src/firecracker/src/api_server_adapter.rs

```diff
@@ -123,7 +123,8 @@ pub(crate) fn run_with_api(
     instance_info: InstanceInfo,
     process_time_reporter: ProcessTimeReporter,
     boot_timer_enabled: bool,
-    payload_limit: Option<usize>,
+    api_payload_limit: usize,
+    mmds_size_limit: usize,
```

Comment:
At the moment, I really don't think we need to. Personally I'd find it a bit more confusing to couple those limits (since semantically they are not coupled).

I'd say to keep things as they are and if we find ourselves in the need to add another limit here, group them somehow, WDYT?

Reply:
Okay, I don't have a very strong preference. We can keep it as is.